ruby_programming/testing_with_rspec/introduction_to_rspec.md

```diff
@@ -0,0 +1,236 @@
+
+Getting this method to "green" shouldn't be too difficult. RSpec clearly provides a reason for the failure: it expected the output to be `7` when we provided the method with `(5, 2)` as the parameters. Instead, it returned `nil`. Why might that be? Well, our `#add` *does* take two parameters...but it does nothing with them! Add the minimum amount of code necessary to get your test to pass:
+
+```ruby
+#lib/calculator.rb
+
+class Calculator
+  def add(a,b)
+    a + b         #=> add this
+  end
+end
+```
+
+Then, run the test again to get a single dot, letting you know that your test has passed:
+
+```bash
+.
+
+Finished in 0.0032 seconds (files took 0.14864 seconds to load)
+1 example, 0 failures
+```
+
+At this point, refactoring isn't necessary. The `#add` method is essentially a one line method. As you progress in your Ruby learning, however, you might find your methods getting more complex, and you might find that you have to make extra efforts to abide by [SOLID](https://en.wikipedia.org/wiki/SOLID_(object-oriented_design)) principles. When that time comes, using RSpec and the "red-green-refactor" cycle will allow you to code with confidence, knowing that your classes and their behaviors continue to meet your specified expectations.
+
+## Assignments
+It's time to put your newfound knowledge to good use. Let's break our `Calculator` test.
+
+1. Let's implement a new test case for your `#add` method, written out for you below. Run the test to see the failure. Write the minimum code necessary to get **both** tests to pass, then refactor if necessary. 
+
+```ruby
+#spec/calculator_spec.rb
+
+describe Calculator do
+  describe "#add" do
+    it "returns the sum of two numbers" do
+      #removed for brevity
+    end
+
+    #add this
+    it "returns the sum of more than two numbers" do
+      calculator = Calculator.new
+      expect(calculator.add(2, 5, 7)).to eql(14)
+    end
+  end
+end
+```
+
+2. Write a test for a new `Calculator` method (`#multiply`, `#subtract`, or `#divide`) using a new `describe` block. Include at least one `it` block with an appropriate expectation clause. Get it to pass, and refactor if necessary.
+3. In the terminal, try running your failing or passing tests with `rspec --format documentation`. What's different?
+4. RSpec reads command line configurations from `.rspec`, one of the two files generated when RSpec is initialized in a project. If you liked the output you got with `--format documentation`, you can use the `.rspec` file to hold that flag. In doing so, you won't have to type it in every time you run your test suite. Open the file in your text editor and, on a new line, add `--format documentation`. For more information on configuring RSpec, see the docs [here](https://relishapp.com/rspec/rspec-core/v/3-7/docs/configuration).
+
+## Additional Resources
+*This section contains helpful links to other content. It isn't required, so consider it supplemental for if you need to dive deeper into something*
```

Comment:
* Solidify these concepts with a [shamelss plug](https://medium.com/@mindovermiles262/getting-started-with-rspec-part-1-9418909f5e53) from another Odin Project contributor

Reply:
Nice! 😄

Reply:
haha. I concur on the shameless plug :)

Reply:
:joy: :+1: